src/idl_gen_general.cpp

```diff
@@ -203,8 +203,13 @@ class GeneralGenerator : public BaseGenerator {
       const std::string &classcode, bool needs_includes) {
     if (!classcode.length()) return true;
 
-    std::string code;
-    code = code + "// " + FlatBuffersGeneratedWarning();
+    std::string code("//------------------------------------------------------------------------------\n"
```

Comment:
There is no need for `----` in the header. The compiler is just looking for `<autogenerated` or `<auto-generated` in the first line comment (which can span multiple lines). The fact that code generators typically use a proper XML tag with a matching closing tag is just a convention.